File: src/prod/src/managed/Api/src/System/Fabric/Common/StartNodeDescriptionKind.cs
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

namespace System.Fabric.Common
{
    using System.Fabric.Interop;

    /// <summary>
    /// This is an enum used to indicate what overload of the RestartNode API is being used 
    /// </summary>
    /// <remarks>
    /// The values indicate whether RestartNode .API is being called using NodeName etc. or using ReplicaSelector etc.
    /// </remarks>
    [Serializable]
    internal enum StartNodeDescriptionKind
    {
        Invalid = NativeTypes.FABRIC_START_NODE_DESCRIPTION_KIND.FABRIC_START_NODE_DESCRIPTION_KIND_INVALID,
        /// <summary>
        /// Do not verify the completion of the action.
        /// </summary>
        UsingNodeName = NativeTypes.FABRIC_START_NODE_DESCRIPTION_KIND.FABRIC_START_NODE_DESCRIPTION_KIND_USING_NODE_NAME,
        /// <summary>
        /// Verify the completion of the action.
        /// </summary>
        UsingReplicaSelector = NativeTypes.FABRIC_START_NODE_DESCRIPTION_KIND.FABRIC_START_NODE_DESCRIPTION_KIND_USING_REPLICA_SELECTOR
    }
}

File: src/prod/src/ServiceModel/management/FaultAnalysisService/StartNodeDescriptionKind.cpp
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

#include "stdafx.h"

namespace Management
{
    namespace FaultAnalysisService
    {
        namespace StartNodeDescriptionKind
        {
            void WriteToTextWriter(Common::TextWriter & w, Enum const & val)
            {
                switch (val)
                {
                case UsingNodeName:
                    w << L"UsingNodeName";
                    return;
                case UsingReplicaSelector:
                    w << L"UsingReplicaSelector";
                    return;
                default:
                    w << Common::wformatString("Unknown StartNodeDescriptionKind value {0}", static_cast<int>(val));
                }
            }

            Enum FromPublicApi(FABRIC_START_NODE_DESCRIPTION_KIND const & kind)
            {
                switch (kind)
                {
                case FABRIC_START_NODE_DESCRIPTION_KIND_USING_NODE_NAME:
                    return UsingNodeName;

                case FABRIC_START_NODE_DESCRIPTION_KIND_USING_REPLICA_SELECTOR:
                    return UsingReplicaSelector;
                default:
                    return None;
                }
            }

            FABRIC_START_NODE_DESCRIPTION_KIND ToPublicApi(Enum const & kind)
            {
                switch (kind)
                {
                case UsingNodeName:
                    return FABRIC_START_NODE_DESCRIPTION_KIND_USING_NODE_NAME;

                case UsingReplicaSelector:
                    return FABRIC_START_NODE_DESCRIPTION_KIND_USING_REPLICA_SELECTOR;

                default:
                    return FABRIC_START_NODE_DESCRIPTION_KIND_INVALID;
                }
            }
        }
    }
}
	
